libs/frontend/feature-build/src/lib/create-code-editor-view.ts

```diff
@@ -42,21 +30,20 @@ export function createCodeEditorView(
       const { textContent } = newState.doc
       if (newState.doc.content.firstChild?.type.name !== 'code_block') {
         // When using hljs with ProseMirror editor, the code must be wrapped inside
-        // `<pre><code>...</code></pre>` for highlighting&styling feature to work. If a user
-        // removes all the text from the editor (i.e. select all text then press backspace),
+        // `<pre><code>...</code></pre>`, which corresponds to the `code_block` node
+        // in the schema for highlighting&styling feature to work.
+        // If a user removes all the text from the editor (i.e. select all text then press backspace),
         // the leading&ending <pre><code> well also be removed, causing the highlighting & styling
-        // feature to stop working. We need to manually wrap the content inside <pre><code> in such cases.
-
-        // FIXME
-        // 1. The current solution will clear the history because it creates a brand-new editor state.
-        //    Find a way to replace the content in the editor state while keeping everything.
-        // 2. New line characters are ignored if the text is copied from a source other than ProseMirror
-        //    editor. For example, VSCode.
-        newState = prosemirrorSetup({
-          schema: schemaCode,
-          content: `<pre><code>${textContent}</code></pre>`,
-          plugins: [...codeEditorPlugins],
-        })
+        // feature to stop working. So we need to manually add the `code_block` node back in such cases.
+        const codeBlockNode = schemaCode.node(
```

Comment:
This strange workaround is also fixed! I found a way to create a new `code_block` node & insert it back to the editor content so that we don't have to initialize a brand new editor state like we did before. I think edit history (redo/undo) should be working fine after" ctrl + a -> ctrl + v" now!